src/EnergyPlus/Material.cc

```diff
@@ -2756,6 +2758,99 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
+    std::string_view cCurrentModuleObject{"MaterialProperty:VariableAbsorptance"};
+    // Call Input Get routine to retrieve material data
+    state.dataInputProcessing->inputProcessor->getObjectItem(state,
+                                                             cCurrentModuleObject,
+                                                             i,
+                                                             alphas,
+                                                             numAlphas,
+                                                             numbers,
+                                                             numNumbers,
+                                                             IOStat,
+                                                             state.dataIPShortCut->lNumericFieldBlanks,
+                                                             state.dataIPShortCut->lAlphaFieldBlanks,
+                                                             state.dataIPShortCut->cAlphaFieldNames,
+                                                             state.dataIPShortCut->cNumericFieldNames);
+
+    // Load the material derived type from the input data.
+    int MaterNum = UtilityRoutines::FindItemInPtrList(alphas(2), state.dataMaterial->Material);
+    if (MaterNum == 0) {
+        ShowSevereError(state,
+                        format("{}: invalid {} entered={}, must match to a valid Material name.",
+                               cCurrentModuleObject,
+                               state.dataIPShortCut->cAlphaFieldNames(1),
+                               alphas(2)));
+        errorsFound = true;
+        return;
+    }
+    auto *thisMaterial = state.dataMaterial->Material(MaterNum);
+
+    if (thisMaterial->Group != Material::MaterialGroup::RegularMaterial) {
+        ShowSevereError(
+            state,
+            format("{}: Reference Material is not appropriate type for Thermal/Solar Absorptance properties, material={}, must have regular "
+                   "properties (Thermal/Solar Absorptance)",
+                   cCurrentModuleObject,
+                   thisMaterial->Name));
+        errorsFound = true;
+        return;
+    }
+
+    Material::VariableAbsCtrlSignal controlSignal = Material::VariableAbsCtrlSignal::SurfaceTemperature; // default value
+    controlSignal = static_cast<VariableAbsCtrlSignal>(getEnumerationValue(VariableAbsCtrlSignalUC, UtilityRoutines::MakeUPPERCase(alphas(3))));
+    //    init to 0 as GetScheduleIndex returns 0 for not-found schedule or curves
+    int functionIdxThermal = 0;
+    int scheduleIdxThermal = 0;
+    int functionIdxSolar = 0;
+    int scheduleIdxSolar = 0;
+    if (controlSignal == VariableAbsCtrlSignal::Scheduled) {
+        scheduleIdxThermal = ScheduleManager::GetScheduleIndex(state, alphas(5));
+        scheduleIdxSolar = ScheduleManager::GetScheduleIndex(state, alphas(7));
+        if (scheduleIdxThermal + scheduleIdxSolar == 0) {
```

Comment:
Don't replace logic with math. It's not robust and also confusing to the reader. `if (schedIdxThermal == 0 && schedIdxSolar == 0)` is better.

Reply:
good point. change it to two equality checks